pkg/cnb/project_descriptor_test.go

```diff
@@ -243,12 +244,61 @@ value = "valueC"
 name = "keyC"
 value = "valueAnotherC"
 				`), 0644)
+						})
+						it("writes all env var files to the platform dir", func() {
+							assert.Nil(t, cnb.ProcessProjectDescriptor(appDir, descriptorPath, platformDir, logger))
+							checkEnvVar(t, platformDir, "keyA", "valueA")
+							checkEnvVar(t, platformDir, "keyB", "valueB")
+							checkEnvVar(t, platformDir, "keyC", "valueAnotherC")
+						})
 					})
-					it("writes all env var files to the platform dir", func() {
-						assert.Nil(t, cnb.ProcessProjectDescriptor(appDir, descriptorPath, platformDir, logger))
-						checkEnvVar(t, platformDir, "keyA", "valueA")
-						checkEnvVar(t, platformDir, "keyB", "valueB")
-						checkEnvVar(t, platformDir, "keyC", "valueAnotherC")
+
+					when("vars where set with new v0.2 project.toml", func() {
+						it.Before(func() {
+							ioutil.WriteFile(projectToml, []byte(`
+[_]
+schema-version = "0.2"
+[[io.buildpacks.build.env]]
+name = "keyA"
+value = "valueA"
+[[io.buildpacks.build.env]]
+name = "keyB"
+value = "valueB"
+[[io.buildpacks.build.env]]
+name = "keyC"
+value = "valueC"
+# check that later keys override previous ones
+[[io.buildpacks.build.env]]
+name = "keyC"
+value = "valueAnotherC"
+				`), 0644)
+						})
+						it("writes all env var files to the platform dir", func() {
+							assert.Nil(t, cnb.ProcessProjectDescriptor(appDir, descriptorPath, platformDir, logger))
+							checkEnvVar(t, platformDir, "keyA", "valueA")
+							checkEnvVar(t, platformDir, "keyB", "valueB")
+							checkEnvVar(t, platformDir, "keyC", "valueAnotherC")
+						})
+					})
+
+					when("vars where set with both versions of v0.2 project.toml", func() {
+						it.Before(func() {
+							ioutil.WriteFile(projectToml, []byte(`
+[_]
+schema-version = "0.2"
+[[io.buildpacks.env]]
```

Comment:
@jjbustamante shouldn't this be io.buildpacks.env.build?

Reply:
That one is defined in L293 see [here](https://github.com/pivotal/kpack/blob/0e3c15f9d5230ff454dbc1776362e7a3f6c7a966/pkg/cnb/project_descriptor_test.go#L289-L295)

In that case, I defined the same variable twice:

```toml
schema-version = "0.2"
[[io.buildpacks.env]]
name = "keyA"
value = "valueA"
# check that new declaration must have higher precedence
[[io.buildpacks.build.env]]
name = "keyA"
value = "newValueA"
```

And then in the assert, I verify the one taken was `newValueA`

Reply:
@jjbustamante I meant to comment on the key itself. Left some notes at https://github.com/pivotal/kpack/pull/1027#pullrequestreview-1095151602